### PR TITLE
docs: update swift sdk

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary
1. add the screen view definition
2. update the definition of session
3. update the definition of user engagement
4. update the example event for record an event
5. add the data type for each attribute
6. add reference link
7. adjust the content structure
8. update the error code
9. fix typo

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: docs/en/sdk-manual/swift.md [hunk 1]
```diff
@@ -10,19 +10,20 @@ The SDK is based on the Amplify for Swift SDK Core Library and developed accordi
 
 Clickstream Swift SDK supports iOS 13+.
 
-[**API Documentation**](https://awslabs.github.io/clickstream-swift/)
+Clickstream requires Xcode 13.4 or higher to build.
```
Comment (zxkane) [inline docs/en/sdk-manual/swift.md:13]: add `Swift SDK`

**Review by zxkane — CHANGES_REQUESTED** [review]

File: docs/en/sdk-manual/swift.md [hunk 2]
```diff
@@ -250,72 +475,85 @@ In Clickstream Swift SDK, we define the `user_engagement` as the app is in the f
...
+        "event_category": "recommended",
+		    "currency": "CNY",
         "_session_duration": 15349,
         "_session_id": "0E6614B7-20230526-062238846",
         "_session_number": 3,
         "_session_start_timestamp": 1685082158847,
-        "product_category": "men's clothing",
-        "product_id": 1
+        "_screen_name": "ProductDetailViewController",
+		    "_screen_unique_id": "5260751568"
```
Comment (zxkane) [inline docs/en/sdk-manual/swift.md:485]: indentation is wrong

**Review by zxkane — CHANGES_REQUESTED** [review]

File: docs/en/sdk-manual/swift.md (hunk 1 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/swift.md:13]: added

File: docs/en/sdk-manual/swift.md [hunk 3]
```diff
@@ -65,64 +66,264 @@ Your `appId` and `endpoint` are already set up in it, here's an explanation of e
...
-    // Override point for customization after application launch.
-    do {
-        try ClickstreamAnalytics.initSDK()
-    } catch {
-        assertionFailure("Fail to initialize ClickstreamAnalytics: \(error)")
+=== "Swift"
+    ```swift
+    import Clickstream
+    
+    ```
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/swift.md:73]: removed

File: docs/en/sdk-manual/swift.md [hunk 4]
```diff
@@ -146,72 +347,96 @@ Clickstream Swift SDK supports the following data types:
 
 3. The event name and attribute name are in case-sensitive, So the event `Add_to_cart` and `add_to_cart` will be Recognized as two different event.
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/swift.md:348]: fixed

File: docs/en/sdk-manual/swift.md (hunk 2 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/swift.md:485]: fixed

**Review by zxkane — APPROVED** [review]

**Pull request merged**